air/src/proof/mod.rs

```diff
@@ -207,61 +207,126 @@ fn get_conjectured_security(
     )
 }
 
-#[cfg(feature = "std")]
 /// Estimates proven security level for the specified proof parameters.
+#[cfg(feature = "std")]
 fn get_proven_security(
     options: &ProofOptions,
     base_field_bits: u32,
     lde_domain_size: u64,
     trace_domain_size: u64,
     collision_resistance: u32,
 ) -> u32 {
-    let extension_field_bits = (base_field_bits * options.field_extension().degree()) as f64;
+    let m_min: usize = 3;
+    let m_max = upper_m(trace_domain_size);
+    let m_optimal = (m_min as u32..m_max as u32)
+        .max_by_key(|&a| {
+            proven_security_protocol_m(
+                options,
+                base_field_bits,
+                lde_domain_size,
+                trace_domain_size,
+                a as usize,
+            )
+        })
+        .expect(
+            "Should not fail since m_max is larger than m_min
+                             for all trace sizes of length greater than 4",
```

Comment:
minor nit: I'd probably put the whole string on a single line (otherwise, error formatting could look weird).

Reply:
Done!